File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length-1;
        int[] ans = new int[2];

        while(l<r){
            while (numbers[l]+numbers[r]< target) l++;
            while (numbers[l]+numbers[r]> target) r--;
            if (numbers[l]+numbers[r]== target){
                ans[0] = l+1;
                ans[1] = r+1;
                break;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nmap = {}
        for i,x in enumerate(nums):
            diff = target - x
            if diff in nmap:
                return [nmap[diff], i]
            nmap[x] = i